crates/rpc/rpc-eth-api/src/core.rs

```diff
@@ -19,6 +23,95 @@ use crate::helpers::{
     EthTransactions, FullEthApi,
 };
 
+/// Network specific `eth` API types.
+pub trait EthApiTypes: Send + Sync {
+    /// Extension of [`EthApiError`], with network specific errors.
+    type Error: IntoRpcError + FromEthApiError + AsEthApiError + FromEvmError;
```

Comment:
could we replace the IntoRpcError with an `Into<ErrorObject>` the toRpcError trait was only necessary so that we can do Into ErrorObject for EthApi